### PR TITLE
Fix XDG spec implementation

### LINKED ISSUE #156: XDG configuration file path implemented incorrectly
> $XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

cf. https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html

Therefore, a lookup path of 
1. `$XDG_CONFIG_HOME/pt/config.toml`
2. `$HOME/.ptconfig.toml`
3. `./.ptconfig.toml`

is wrong. The second one should be `$HOME/.config/pt/config.toml`.

### PR DESCRIPTION
**Pull request by vially**
If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used

Fix #156

### CODE REVIEW

File: platinum_searcher.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func (p PlatinumSearcher) Run(args []string) int {
 	conflag.LongHyphen = true
 	conflag.BoolValue = false
 	for _, c := range []string{
-		filepath.Join(os.Getenv("XDG_CONFIG_HOME"), "pt", "config.toml"),
+		filepath.Join(xdgConfigHomeDir(), "pt", "config.toml"),
 		filepath.Join(home.Dir(), ".ptconfig.toml"),
```
Comment (maletor) [inline platinum_searcher.go]: Can we remove this one?

Comment (vially) [inline platinum_searcher.go]: That would break compatibility for users who are currently using `$HOME/.ptconfig.toml` so I'm not sure that's a good idea.

**Comment by maletor** [comment]
👍

**Comment by monochromegane** [comment]
Thanks!! LGTM 👍

**Pull request merged**